x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupField.java

```diff
@@ -36,6 +37,9 @@
     public static final String ROLLUP_MISSING = "ROLLUP_MISSING_40710B25931745D4B0B8B310F6912A69";
     public static final List<String> SUPPORTED_METRICS = Arrays.asList(MaxAggregationBuilder.NAME, MinAggregationBuilder.NAME,
```

Comment:
I wonder if we should have `SUPPORTED_NUMERIC_METRICS`, `SUPPORTED_DATE_METRICS`, and then a union of those in a third list `SUPPORTED_METRICS`?

E.g. in the MetricsConfig constructor (and a few other places), we validate that an agg is one of the supported metrics.  We don't yet have the field mapper type, so we can't validate numeric vs date field and their associated metrics, but it's a quick check to make sure the config is at least sane.

The current `SUPPORTED_METRICS` is technically a superset of the date metrics too so everything is fine.  But I'm worried in the future the two lists may diverge for whatever reason, and we miss the validation because dates support something numerics don't, or vice versa.

Reply:
So, I am making `SUPPORTED_NUMERIC_METRICS` and `SUPPORTED_DATE_METRICS` two separate lists with overlapping metrics. 

Then there is a `Set<String>` that will be `SUPPORTED_METRICS` that is a union of the two (need to verify that the type change will not break other logic). This should lay the ground work/pattern for future non-overlapping supported metrics.